docs/website/docs/general-usage/credentials/config_providers.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Configuration Providers
+description: Configuration dlt Providers
+keywords: [credentials, secrets.toml, secrets, config, configuration, environment
+      variables, provider]
+---
+
+# Configuration Providers
+## The provider hierarchy
+
+If function signature has arguments that may be injected, `dlt` looks for the argument values in
```

Comment:
As in `docs/website/docs/general-usage/credentials/configuration.md`, to welcome the reader it's best to begin the narrative with a brief summary or description of what the concept is. What is a configuration provider?